packages/react-test-renderer/src/ReactShallowRenderer.js

```diff
@@ -366,6 +367,13 @@ class ReactShallowRenderer {
       return fn;
     };
 
+    const useEffect = (fn: Function, deps: Array<mixed> | void | null): Function => {
+      this._validateCurrentlyRenderingComponent();
+      this._createWorkInProgressHook();
+
+      //fill in impl
```

Comment:
This is the part I'm trying to still figure out how to do. I think I have a general idea of how things work, but before I went any further I wanted to confirm that this makes sense.